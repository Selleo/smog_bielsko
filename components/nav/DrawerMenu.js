import React, { Component, PropTypes } from 'react';
import { StyleSheet, View } from 'react-native';

import DrawerCityLink from './DrawerCityLink'
import DrawerHeader from './DrawerHeader'

export default class DrawerMenu extends Component {
  render() {
    return (
      <View style={styles.container}>
        <DrawerHeader text='Wybierz miejscowość:' />
        <DrawerCityLink getCurrentStationId={this.props.getCurrentStationId} station={{id: 6535, name: 'Bielsko-Biała'}} closeDrawer={this.props.closeDrawer} />
        <DrawerCityLink getCurrentStationId={this.props.getCurrentStationId} station={{id: 8281, name: 'Żywiec'}} closeDrawer={this.props.closeDrawer} />
        <DrawerCityLink getCurrentStationId={this.props.getCurrentStationId} station={{id: 6548, name: 'Ustroń'}} closeDrawer={this.props.closeDrawer} />
        <DrawerCityLink getCurrentStationId={this.props.getCurrentStationId} station={{id: 6547, name: 'Tychy'}} closeDrawer={this.props.closeDrawer} />
        <DrawerCityLink getCurrentStationId={this.props.getCurrentStationId} station={{id: 6540, name: 'Katowice'}} closeDrawer={this.props.closeDrawer} />
        <DrawerCityLink getCurrentStationId={this.props.getCurrentStationId} station={{id: 6544, name: 'Wodzisław Śląski'}} closeDrawer={this.props.closeDrawer} />
        <DrawerCityLink getCurrentStationId={this.props.getCurrentStationId} station={{id: 6542, name: 'Rybnik'}} closeDrawer={this.props.closeDrawer} />
        <DrawerCityLink getCurrentStationId={this.props.getCurrentStationId} station={{id: 2793, name: 'Biały Krzyż (Czechy)'}} closeDrawer={this.props.closeDrawer} />
      </View>
    )
  }
  static propTypes = {
    closeDrawer: PropTypes.func.isRequired,
    getCurrentStationId: PropTypes.func.isRequired,
  };
}

const styles = StyleSheet.create({
  container: {
    paddingTop: 30
  }
});
