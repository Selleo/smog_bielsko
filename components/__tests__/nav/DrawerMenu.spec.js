import 'react-native';
import React                          from 'react';

import { shallow }                    from 'enzyme';
import chai, { expect }               from 'chai';
import dirtyChai                      from 'dirty-chai';

import DrawerMenu                 from '../../nav/DrawerMenu';

const originalExpect = global.expect;

chai.use(dirtyChai);

let wrapper, props;

describe('DrawerMenu', () => {
  beforeEach(() => {
    props = {
      closeDrawer: () => true,
      getCurrentStationId: () => true
    };

    wrapper = shallow(<DrawerMenu {...props}/>);
  });

  it('renders correctly', () => {
    originalExpect(wrapper).toMatchSnapshot();
    expect(wrapper).to.exist();
  });

  it('render staton name on html', () => {
    expect(wrapper.children().nodes.length).to.equal(9)
  });
});
