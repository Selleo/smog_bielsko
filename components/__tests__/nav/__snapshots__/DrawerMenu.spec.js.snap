exports[`DrawerMenu renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "paddingTop": 30,
          }
    }>
    <DrawerHeader
        text="Wybierz miejscowość:" />
    <DrawerCityLink
        closeDrawer={[Function]}
        getCurrentStationId={[Function]}
        station={
            Object {
                "id": 6535,
                "name": "Bielsko-Biała",
              }
        } />
    <DrawerCityLink
        closeDrawer={[Function]}
        getCurrentStationId={[Function]}
        station={
            Object {
                "id": 8281,
                "name": "Żywiec",
              }
        } />
    <DrawerCityLink
        closeDrawer={[Function]}
        getCurrentStationId={[Function]}
        station={
            Object {
                "id": 6548,
                "name": "Ustroń",
              }
        } />
    <DrawerCityLink
        closeDrawer={[Function]}
        getCurrentStationId={[Function]}
        station={
            Object {
                "id": 6547,
                "name": "Tychy",
              }
        } />
    <DrawerCityLink
        closeDrawer={[Function]}
        getCurrentStationId={[Function]}
        station={
            Object {
                "id": 6540,
                "name": "Katowice",
              }
        } />
    <DrawerCityLink
        closeDrawer={[Function]}
        getCurrentStationId={[Function]}
        station={
            Object {
                "id": 6544,
                "name": "Wodzisław Śląski",
              }
        } />
    <DrawerCityLink
        closeDrawer={[Function]}
        getCurrentStationId={[Function]}
        station={
            Object {
                "id": 6542,
                "name": "Rybnik",
              }
        } />
    <DrawerCityLink
        closeDrawer={[Function]}
        getCurrentStationId={[Function]}
        station={
            Object {
                "id": 2793,
                "name": "Biały Krzyż (Czechy)",
              }
        } />
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "paddingTop": 30,
                }
      }>
      <DrawerHeader
            text="Wybierz miejscowość:" />
      <DrawerCityLink
            closeDrawer={[Function]}
            getCurrentStationId={[Function]}
            station={
                  Object {
                        "id": 6535,
                        "name": "Bielsko-Biała",
                      }
            } />
      <DrawerCityLink
            closeDrawer={[Function]}
            getCurrentStationId={[Function]}
            station={
                  Object {
                        "id": 8281,
                        "name": "Żywiec",
                      }
            } />
      <DrawerCityLink
            closeDrawer={[Function]}
            getCurrentStationId={[Function]}
            station={
                  Object {
                        "id": 6548,
                        "name": "Ustroń",
                      }
            } />
      <DrawerCityLink
            closeDrawer={[Function]}
            getCurrentStationId={[Function]}
            station={
                  Object {
                        "id": 6547,
                        "name": "Tychy",
                      }
            } />
      <DrawerCityLink
            closeDrawer={[Function]}
            getCurrentStationId={[Function]}
            station={
                  Object {
                        "id": 6540,
                        "name": "Katowice",
                      }
            } />
      <DrawerCityLink
            closeDrawer={[Function]}
            getCurrentStationId={[Function]}
            station={
                  Object {
                        "id": 6544,
                        "name": "Wodzisław Śląski",
                      }
            } />
      <DrawerCityLink
            closeDrawer={[Function]}
            getCurrentStationId={[Function]}
            station={
                  Object {
                        "id": 6542,
                        "name": "Rybnik",
                      }
            } />
      <DrawerCityLink
            closeDrawer={[Function]}
            getCurrentStationId={[Function]}
            station={
                  Object {
                        "id": 2793,
                        "name": "Biały Krzyż (Czechy)",
                      }
            } />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <DrawerMenu
        closeDrawer={[Function]}
        getCurrentStationId={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": DrawerMenu {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "closeDrawer": [Function],
          "getCurrentStationId": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "paddingTop": 30,
                            }
          }>
          <DrawerHeader
                    text="Wybierz miejscowość:" />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 6535,
                                        "name": "Bielsko-Biała",
                                      }
                    } />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 8281,
                                        "name": "Żywiec",
                                      }
                    } />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 6548,
                                        "name": "Ustroń",
                                      }
                    } />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 6547,
                                        "name": "Tychy",
                                      }
                    } />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 6540,
                                        "name": "Katowice",
                                      }
                    } />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 6544,
                                        "name": "Wodzisław Śląski",
                                      }
                    } />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 6542,
                                        "name": "Rybnik",
                                      }
                    } />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 2793,
                                        "name": "Biały Krzyż (Czechy)",
                                      }
                    } />
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Object {
                              "paddingTop": 30,
                            }
          }>
          <DrawerHeader
                    text="Wybierz miejscowość:" />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 6535,
                                        "name": "Bielsko-Biała",
                                      }
                    } />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 8281,
                                        "name": "Żywiec",
                                      }
                    } />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 6548,
                                        "name": "Ustroń",
                                      }
                    } />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 6547,
                                        "name": "Tychy",
                                      }
                    } />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 6540,
                                        "name": "Katowice",
                                      }
                    } />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 6544,
                                        "name": "Wodzisław Śląski",
                                      }
                    } />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 6542,
                                        "name": "Rybnik",
                                      }
                    } />
          <DrawerCityLink
                    closeDrawer={[Function]}
                    getCurrentStationId={[Function]}
                    station={
                              Object {
                                        "id": 2793,
                                        "name": "Biały Krzyż (Czechy)",
                                      }
                    } />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DrawerMenu
    closeDrawer={[Function]}
    getCurrentStationId={[Function]} />,
}
`;
